core/src/Streamly/Data/Array/Generic/Mut.hs

```diff
@@ -0,0 +1,63 @@
+-- |
+-- Module      : Streamly.Data.Array.Generic.Mut
+-- Copyright   : (c) 2020 Composewell Technologies
+-- License     : BSD3-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+module Streamly.Data.Array.Generic.Mut
+(
+    -- * Type
+    -- $arrayNotes
+      Array (..)
+
+    -- * Constructing and Writing
+    -- ** Construction
+    -- *** Uninitialized Arrays
+    , newArray
+
+    -- *** From streams
+    , writeNUnsafe
```

Comment:
Do not expose everything from the internal module. Please see what is exposed from other array modules and expose only carefully reviewed APIs that are consistent across all array modules.